crates/networking/rpc/engine/payload.rs

```diff
@@ -355,23 +355,45 @@ impl GetPayloadRequest {
 
     fn handle(&self, context: RpcApiContext) -> Result<Value, RpcErr> {
         info!("Requested payload with id: {:#018x}", self.payload_id);
-        let Some(mut payload) = context.storage.get_payload(self.payload_id)? else {
+
+        let payload = context.storage.get_payload(self.payload_id)?;
+
+        let Some((mut payload_block, block_value, blobs_bundle, closed)) = payload else {
```

Comment:
make sure you're not swallowing errors here

Reply:
I'll take a look!

Reply:
it looks good though, I think the problem is when there are results.